src/Microsoft.NetCore.Analyzers/Core/ImmutableCollections/DoNotCallToImmutableCollectionOnAnImmutableCollectionValue.Fixer.cs

```diff
@@ -19,10 +26,70 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        private static readonly ImmutableArray<string> ToImmutableMethodNames = new[]
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
+            "ToImmutableArray",
+            "ToImmutableDictionary",
+            "ToImmutableHashSet",
+            "ToImmutableList",
+            "ToImmutableSortedDictionary",
+            "ToImmutableSortedSet"
+        }.ToImmutableArray();
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Diagnostic diagnostic = context.Diagnostics.FirstOrDefault();
+            if (diagnostic == null)
+            {
+                return;
+            }
+
+            Document document = context.Document;
+            TextSpan span = context.Span;
+            SyntaxNode root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode invocationNode = root.FindNode(span, getInnermostNodeForTie: true);
+            if (invocationNode == null)
+            {
+                return;
+            }
+
+            SemanticModel semanticModel = await document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var invocationOp = semanticModel.GetOperation(invocationNode) as IInvocationOperation;
+            if (invocationOp == null || !ToImmutableMethodNames.Contains(invocationOp.TargetMethod.Name))
+            {
+                return;
+            }
+
+            string title = SystemCollectionsImmutableAnalyzersResources.RemoveRedundantCall;
+
+            context.RegisterCodeFix(new MyCodeAction(title,
+                                        async cancellationToken => await RemoveRedundantCall(document, root, invocationNode, invocationOp).ConfigureAwait(false),
+                                        equivalenceKey: title),
+                                    diagnostic);
+        }
+
+        private static Task<Document> RemoveRedundantCall(Document document, SyntaxNode root, SyntaxNode invocationNode, IInvocationOperation invocationOp)
+        {
+            SyntaxNode instance = GetInstance(invocationOp).WithTrailingTrivia(invocationNode.GetTrailingTrivia());
```

Comment:
Using `WithTriviaFrom` instead to preserve both leading and trialing trivia?

Reply:
@genlu sorry, is that a question? Or a suggestion?

Reply:
I'd suggest using it unless you have a reason to ignore leading trivia

---
In reply to: [268350158](https://github.com/dotnet/roslyn-analyzers/pull/2238#discussion_r268350158) [](ancestors = 268350158)